### PR TITLE
Refactoring variables

### PR DESCRIPTION
**Pull request by maxcapodi78**
Refactoring of parent variables
refactoring of AEDT Object initialization

### CODE REVIEW

File: pyaedt/edb.py [hunk 1]
```diff
@@ -772,7 +772,7 @@ def close_edb(self):
         end = time.time()-start_time
         self._messenger.add_info_message("Release Time {}".format(end))
         while gc.collect() != 0:
```
Comment (akaszynski) [inline pyaedt/edb.py:774]: You may want to implement some sort of a timeout here.

File: pyaedt/application/AnalsyisEmit.py [hunk 2]
```diff
@@ -68,7 +68,7 @@ def __init__(
             student_version,
         )
         self._modeler = ModelerEmit(self)
-        self._post = PostProcessor(self)
+        #self._post = PostProcessor(self)
```
Comment (MaxJPRey) [inline pyaedt/application/AnalsyisEmit.py:71]: This line can be removed.

File: pyaedt/application/Design.py [hunk 3]
```diff
@@ -549,6 +549,19 @@ def __init__(
...
+
+        self.oimportexport = None
+
+        self.ooptimetrics = None
+
+        self.ooutput_variable = None
+
+        self.odesktop = None
+
+        self.oanalysis = None
```
Comment (MaxJPRey) [inline pyaedt/application/Design.py:564]: @maxcapodi78 Maybe we don't need to have new lines between each data member assignation. Your call.

File: pyaedt/application/Design.py [hunk 4]
```diff
@@ -594,7 +611,12 @@ def __delitem__(self, key):
 
     @property
     def logger(self):
-        "Logger for the Design."
+        """Logger for the Design.
+
+        Returns
+        -------
+        :class:`pyaedt.aedt_logger.AedtLogger`
+        """
```
Comment (MaxJPRey) [inline pyaedt/application/Design.py:619]: 👍

File: pyaedt/edb.py [hunk 5]
```diff
@@ -741,6 +730,63 @@ def edb_value(self, val):
...
+            return False
+
+    @aedt_exception_handler
+    def _wait_for_file_release(self, timeout=30, file_to_release=None):
+        if not file_to_release:
+            file_to_release = os.path.join(self.edbpath)
+        tstart = time.time()
+        while True:
+            if self._is_file_existing_and_released(file_to_release):
+                # print 'File is released'
```
Comment (MaxJPRey) [inline pyaedt/edb.py:762]: The commented `print`can either be deleted or replaced by writting down the message into the log.

File: pyaedt/edb.py [hunk 6]
```diff
@@ -937,43 +986,54 @@ def create_cutout(
...
-                retry_ntimes(
-                    10,
-                    self.layout_methods.InitializeBuilder,
-                    self._db,
-                    self._active_cell,
-                    self.edbpath,
-                    self.edbversion,
-                    self.standalone,
-                )
+                dllpath = os.path.join(os.path.dirname(__file__), "dlls", "EDBLib")
```
Comment (MaxJPRey) [inline pyaedt/edb.py:1013]: `dllpath` can be renamed such as `dll_path`.

File: pyaedt/edb.py [hunk 7]
```diff
@@ -937,43 +986,54 @@ def create_cutout(
...
+                    self.layout_methods.LoadDataModel(dllpath, self.edbversion)
+                except:
+                    pass
+                # self.layout_methods.InitializeBuilder(
+                #                                     self._db,
+                #                                     self._active_cell,
+                #                                     self.edbpath,
+                #                                     self.edbversion,
+                #                                     self.standalone,
+                #                                 )
```
Comment (MaxJPRey) [inline pyaedt/edb.py:1024]: To be deleted.

File: pyaedt/edb.py [hunk 8]
```diff
@@ -1064,30 +1122,34 @@ def create_cutout_on_point_list(
         _dbCells = [_cutout]
         if output_aedb_path:
             db2 = self.edb.Database.Create(output_aedb_path)
-            # Function input is the name of a .aedb folder inside which the edb.def will be created.
-            # Ex: 'D:/backedup/EDB/TEST PROJECTS/CUTOUT/N1.aedb'
+            db2.Save()
```
Comment (MaxJPRey) [inline pyaedt/edb.py:1125]:
```suggestion
            _success = db2.Save()
```

File: pyaedt/edb_core/EDB_Data.py [hunk 9]
```diff
@@ -381,14 +381,14 @@ class EDBLayers(object):
...
     Parameters
     ----------
-    parent :
+    edbstackup :
         Inherited AEDT object.
 
     """
 
-    def __init__(self, parent):
+    def __init__(self, edbstackup):
```
Comment (MaxJPRey) [inline pyaedt/edb_core/EDB_Data.py:389]: `edbstackup `can be renamed as `edb_stackup`

File: pyaedt/edb_core/EDB_Data.py [hunk 10]
```diff
@@ -1280,8 +1280,8 @@ def material(self, materialname):
 class EDBPinInstances(object):
     """Manages EDB functionalities in instances."""
 
-    def __init__(self, parent, pin):
-        self.parent = parent
+    def __init__(self, edbcomponents, pin):
```
[the remark this answers is not in the document]
Comment (maxcapodi78) [reply pyaedt/edb_core/EDB_Data.py:1283]: done

File: pyaedt/edb.py [hunk 11]
```diff
@@ -751,15 +797,18 @@ def close_edb(self):
...
         """
-        time.sleep(1)
+        time.sleep(2)
         self._db.Close()
-        self._messenger.add_info_message("Database successfully closed.")
-        time.sleep(1)
+        time.sleep(2)
+        start_time = time.time()
+        self._wait_for_file_release()
+        end = time.time()-start_time
```
[the remark this answers is not in the document]
Comment (maxcapodi78) [reply pyaedt/edb.py:786]: Done

File: pyaedt/edb_core/padstack.py [hunk 12]
```diff
@@ -176,27 +176,41 @@ def create_padstack(
         str
             Name of the padstack if the operation is successful.
         """
+        var_server_db = self.db.GetVariableServer()
```
Comment (MaxJPRey) [inline pyaedt/edb_core/padstack.py:179]:
```suggestion
        server = self.db.GetVariableServer()
```

File: pyaedt/edb_core/padstack.py [hunk 13]
```diff
@@ -176,27 +176,41 @@ def create_padstack(
...
+        var_names = var_server_db.GetAllVariableNames()
```
Comment (MaxJPRey) [inline pyaedt/edb_core/padstack.py:180]:
```suggestion
        variable_names = var_server_db.GetAllVariableNames()
```

**Pull request merged**